plugin/sampling/strategystore/static/strategy_store.go

```diff
@@ -83,32 +84,76 @@ func (h *strategyStore) Close() {
 	h.cancelFunc()
 }
 
-func (h *strategyStore) autoUpdateStrategies(interval time.Duration, filePath string) {
+func downloadSamplingStrategies(url string) ([]byte, error) {
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, fmt.Errorf("failed to download sampling strategies: %w", err)
+	}
+
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf(
+			"receiving %s while downloading strategies file",
+			resp.Status,
+		)
+	}
+
+	buf := new(bytes.Buffer)
+	if _, err = buf.ReadFrom(resp.Body); err != nil {
+		return nil, fmt.Errorf("failed to read sampling strategies from downloaded JSON: %w", err)
+	}
+	return buf.Bytes(), nil
+}
+
+func isURL(str string) bool {
+	u, err := url.Parse(str)
+	return err == nil && u.Scheme != "" && u.Host != ""
+}
+
+func samplingStrategyLoader(strategiesFile string) func() (string, error) {
+	return func() (string, error) {
+		if ok := isURL(strategiesFile); ok {
+			currBytes, err := downloadSamplingStrategies(strategiesFile)
+			if err != nil {
+				return "", err
+			}
+			return string(currBytes), nil
+		}
+
+		currBytes, err := ioutil.ReadFile(strategiesFile)
+		if err != nil {
+			return "", fmt.Errorf("failed to open sampling strategies file: %w", err)
+		}
+		return string(currBytes), nil
+	}
+}
+
+func (h *strategyStore) autoUpdateStrategies(interval time.Duration, strategiesFile string) {
+	loader := samplingStrategyLoader(strategiesFile)
 	lastValue := ""
 	ticker := time.NewTicker(interval)
 	defer ticker.Stop()
 	for {
 		select {
 		case <-ticker.C:
-			lastValue = h.reloadSamplingStrategyFile(filePath, lastValue)
+			lastValue = h.reloadSamplingStrategy(loader, lastValue)
 		case <-h.ctx.Done():
 			return
 		}
 	}
 }
 
-func (h *strategyStore) reloadSamplingStrategyFile(filePath string, lastValue string) string {
-	currBytes, err := ioutil.ReadFile(filepath.Clean(filePath))
+func (h *strategyStore) reloadSamplingStrategy(loader func() (string, error), lastValue string) string {
+	newValue, err := loader()
 	if err != nil {
-		h.logger.Error("failed to load sampling strategies", zap.String("file", filePath), zap.Error(err))
+		h.logger.Error("failed to load sampling strategies", zap.Error(err))
 		return lastValue
 	}
-	newValue := string(currBytes)
 	if lastValue == newValue {
 		return lastValue
 	}
-	if err = h.updateSamplingStrategy(currBytes); err != nil {
-		h.logger.Error("failed to update sampling strategies from file", zap.Error(err))
+	if err := h.updateSamplingStrategy([]byte(newValue)); err != nil {
```

Comment:
Consider if `([]byte, error)` would be more appropriate signature for the `loader()`, to avoid casting back and forth.